numpy/core/src/common/simd/sse/misc.h

```diff
@@ -0,0 +1,230 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+#ifndef _NPY_SIMD_SSE_MISC_H
+#define _NPY_SIMD_SSE_MISC_H
+
+// vector with zero lanes
+#define npyv_zero_u8  _mm_setzero_si128
+#define npyv_zero_s8  _mm_setzero_si128
+#define npyv_zero_u16 _mm_setzero_si128
+#define npyv_zero_s16 _mm_setzero_si128
+#define npyv_zero_u32 _mm_setzero_si128
+#define npyv_zero_s32 _mm_setzero_si128
+#define npyv_zero_u64 _mm_setzero_si128
+#define npyv_zero_s64 _mm_setzero_si128
+#define npyv_zero_f32 _mm_setzero_ps
+#define npyv_zero_f64 _mm_setzero_pd
+
+// vector with a specific value set to all lanes
+#define npyv_setall_u8(VAL)  _mm_set1_epi8((char)VAL)
+#define npyv_setall_s8(VAL)  _mm_set1_epi8((char)VAL)
+#define npyv_setall_u16(VAL) _mm_set1_epi16((short)VAL)
+#define npyv_setall_s16(VAL) _mm_set1_epi16((short)VAL)
+#define npyv_setall_u32(VAL) _mm_set1_epi32((int)VAL)
+#define npyv_setall_s32(VAL) _mm_set1_epi32(VAL)
+#if !defined(__x86_64__) && !defined(_M_X64)
+    #define npyv_setall_u64(VAL) _mm_set_epi32((int)(VAL >> 32), (int)VAL, (int)(VAL >> 32), (int)VAL)
+    #define npyv_setall_s64 npyv_setall_u64
+#else
+    #define npyv_setall_u64(VAL) _mm_set1_epi64x(VAL)
+    #define npyv_setall_s64(VAL) _mm_set1_epi64x(VAL)
+#endif
+#define npyv_setall_f32(VAL) _mm_set1_ps(VAL)
+#define npyv_setall_f64(VAL) _mm_set1_pd(VAL)
+
+/**
+ * vector with specific values set to each lane and
+ * set a specific value to all remained lanes
+ *
+ * Args that generated by NPYV__SET_FILL_* not going to expand if
+ * _mm_setr_* are defined as macros.
+ */
+NPY_FINLINE __m128i npyv__setr_epi8(
+    char i0, char i1, char i2,  char i3,  char i4,  char i5,  char i6,  char i7,
+    char i8, char i9, char i10, char i11, char i12, char i13, char i14, char i15)
+{
+    return _mm_setr_epi8(i0, i1, i2, i3, i4, i5, i6, i7, i8, i9, i10, i11, i12, i13, i14, i15);
+}
+NPY_FINLINE __m128i npyv__setr_epi16(short i0, short i1, short i2, short i3, short i4, short i5,
+                                     short i6, short i7)
+{
+    return _mm_setr_epi16(i0, i1, i2, i3, i4, i5, i6, i7);
+}
+NPY_FINLINE __m128i npyv__setr_epi32(int i0, int i1, int i2, int i3)
+{
+    return _mm_setr_epi32(i0, i1, i2, i3);
+}
+NPY_FINLINE __m128i npyv__setr_epi64(npy_int64 i0, npy_int64 i1)
+{
+    return _mm_set_epi64x(i1, i0);
```

Comment:
It should be `i0, i1`?

Reply:
no, it's correct since I'm using `_set_` not `_set**r**_` so args passed as little-endian order.
I used the little-endian set since the reversed intrinsic not available in a wide range of compilers.